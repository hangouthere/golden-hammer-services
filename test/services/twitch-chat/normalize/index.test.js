const Clazz = require('@/services/twitch-chat/normalize');

jest.mock('@/services/twitch-chat/normalize/EventNormalizeMap', () => ({
  TestEvent_No_Normalizer: {
    EventClassification: 'TestEvent1.FullyClassified'
  },
  TestEvent_Normalizer: {
    EventClassification: 'TestEvent2.FullyClassified',
    Normalizer: jest.fn(() => ({
      timestamp: 'testTimeStampe',
      normalizedData: 'testNormlizedData'
    }))
  }
}));

describe('Twitch Chat Normalize', () => {
  let clazz;

  beforeEach(() => {
    clazz = new Clazz();
  });

  it("should throw an error if it can't map the event name", () => {
    expect(() => clazz.normalize('missingEventName', { bogus: true })).toThrowError(
      'Invalid Event Type for Processing'
    );
  });

  it("should throw an error if there isn't a Normalizer assigned", () => {
    expect(() => clazz.normalize('TestEvent_No_Normalizer', {})).toThrowError('Normalizer missing');
  });

  it('should delegate normalizing and extract sub-context', () => {
    const map = require('@/services/twitch-chat/normalize/EventNormalizeMap');
    const incomingEventArguments = {
      foo: true,
      bar: false,
      awesome: 'yes'
    };

    const retVal = clazz.normalize('TestEvent_Normalizer', incomingEventArguments);

    expect(map.TestEvent_Normalizer.Normalizer).toHaveBeenCalled();
    expect(retVal).toMatchSnapshot();
  });
});
