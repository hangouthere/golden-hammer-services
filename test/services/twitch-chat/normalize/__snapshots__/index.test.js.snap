// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twitch Chat Normalize should delegate normalizing and extract sub-context 1`] = `
Object {
  "eventClassification": "TestEvent2.FullyClassified",
  "normalizedData": "testNormlizedData",
  "timestamp": "testTimeStamp",
}
`;
